### ISSUE TITLE
complete freeze on opening

### ISSUE THREAD

**Issue by kuldeepdhaka** [openscad/openscad#5240]
**Describe the bug**
OpenSCAD do not work at all when opening the file 'bla.scad'

**To Reproduce**
1. Download both bla.scad and polyround.scad (too obvious to say remove .txt? :)
2. Open bla.scad
3. Amazed?

**Expected behavior**
I should be able to work

**Code reproducing the issue**
Dummy code presented

**Screenshots**

**Environment and Version info (please complete the following information):**
- Linux 127.0.0.1localhost 6.9.1-arch1-2 #1 SMP PREEMPT_DYNAMIC Wed, 22 May 2024 13:47:07 +0000 x86_64 GNU/Linux
 - OpenSCAD version 2021.01

**Additional context**
[polyround.scad.txt](https://github.com/user-attachments/files/16383338/polyround.scad.txt)
[bla.scad.txt](https://github.com/user-attachments/files/16383339/bla.scad.txt)

**Comment by t-paul** [comment]
Enabling Preferences -> Advanced -> "Stop on first warning" guards against such issues.

**Comment by kuldeepdhaka** [comment]
@t-paul I understand your suggestion (thanks!) but as a user I do expect some form of "reliability" when I open a file.

**Comment by thehans** [comment]
FWIW the reason for the slowness (as far as I can recall from last time I looked at these sort of performance issues) is that the "Error Log" handles excessive numbers of warnings/errors extremely poorly.  I think it rebuilds the whole grid for each addition, giving it a O(n^2) kind of performance.

The given code produces over 31K warnings due to undefined values.  You can get through all of them in a flash if you run on the command line
```
$ openscad -o bla.stl bla.scad 2> log.txt
$ wc -l log.txt
31398 log.txt
```

Also see: https://github.com/openscad/openscad/issues/4459 

Adding a hard cap on number of allowable errors before terminating is one suggested solutiont that would allow the UI to finish.

**Comment by pca006132** [comment]
Maybe we can only update the error log once when the evaluation is finished? Or add a cap and switch to this lazy rendering when there are too many errors.

**Comment by thehans** [comment]
Closing as duplicate of #4459
